h5netcdf/core.py

```diff
@@ -95,16 +96,25 @@ def _lookup_dimensions(self):
             return (child_name,)
 
         dims = []
+        phony_dims = Counter()
         for axis, dim in enumerate(self._h5ds.dims):
-            # TODO: read dimension labels even if there is no associated
-            # scale? it's not netCDF4 spec, but it is unambiguous...
-            # Also: the netCDF lib can read HDF5 datasets with unlabeled
-            # dimensions.
-            if len(dim) == 0:
-                raise ValueError('variable %r has no dimension scale '
-                                 'associated with axis %s'
-                                 % (self.name, axis))
-            name = _name_from_dimension(dim)
+            if len(dim):
+                name = _name_from_dimension(dim)
+            else:
+                # if unlabeled dimensions are found
+                if self._root._phony_dim_count is None:
+                    raise ValueError('variable %r has no dimension scale '
+                                     'associated with axis %s. \n'
+                                     'Use phony_dims=%r for sorted naming or '
+                                     'phony_dims=%r for per access naming.'
+                                     % (self.name, axis, 'sort', 'access'))
```

Comment:
Do you have a use-case for "per access" naming?

If not, I would rather leave it out to keep things slightly simpler.

Reply:
Actually this is my use case. While working on this I found that iterating over all groups (with `phony_dims='sort'`) terribly affects performance. When using the `per access` approach this phony dimension detection is deferred to the time when data is accessed. This really improves in cases when working with hundreds or thousands of data files (eg. one day of a single radar of German Weather Service has > 50.000 files).

Reply:
OK, cool. I just wanted to be sure we were adding this feature for a clear reason.